### ISSUE TITLE
Facebook Login not working

### ISSUE THREAD

**Issue by cksagherian** [q2a/question2answer#510]
The problem is causing the facebook login app to successfully login, then redirect to a different page. This isn't working on the main q2a actual site either. Please help this is extremely frustrating. what did facebook change? it was working just a few days ago. 

I currently have it turned off. I've also tried using login master and the same thing happens with google and everyone else. Is this a google/facebook issue?

Thank you in adavnce.

**Comment by cksagherian** [comment]
There also seems to be a refresh loop happening on the main site for some reason as well. I'm wondering if this is related. Thank you

**Comment by iamdand** [comment]
Same thing here. Loops on and on.

**Comment by valan** [comment]
this fix worked for me:
http://stackoverflow.com/questions/42994019/facebook-graph-api-not-work-from-2-2-to-2-3

**Comment by iamdand** [comment]
Worked for me as well. Thanks for the tip, @valan !

**Comment by cksagherian** [comment]
For those who are not code savvy, I've compiled the new php file that should replace.

/public_html/qa-plugin/facebook-login/base_facebook.php

Simply delete or rename original and replace with this new file. 

NOTE**** GitHub only allows .txt files to be uploaded so just rename to .php.

The credit for this fix should be given to @iamdand. I'm waiting to hear back on whether there are any security issues that would cause this to not be used on a "Production Site" or if he simply doesn't want blame for people copying and pasting the wrong code and therefore screwing something up. ;-)

I have verified this to work though.
 
[base_facebook.txt](https://github.com/q2a/question2answer/files/879387/base_facebook.txt)
